src/engraving/dom/cmd.cpp

```diff
@@ -3273,46 +3275,49 @@ void Score::cmdMirrorNoteHead()
 
 void Score::cmdIncDecDuration(int nSteps, bool stepDotted)
 {
-    EngravingItem* el = selection().element();
-    if (el == 0) {
-        return;
-    }
-    if (el->isNote()) {
-        el = el->parentItem();
-    }
-    if (!el->isChordRest()) {
-        return;
-    }
-
-    ChordRest* cr = toChordRest(el);
-
-    // if measure rest is selected as input, then the correct initialDuration will be the
-    // duration of the measure's time signature, else is just the input state's duration
-    TDuration initialDuration;
-    if (cr->durationType() == DurationType::V_MEASURE) {
-        initialDuration = TDuration(cr->measure()->timesig(), true);
-
-        if (initialDuration.fraction() < cr->measure()->timesig() && nSteps > 0) {
-            // Duration already shortened by truncation; shorten one step less
-            --nSteps;
+    if (selection().isRange()) {
+        if (!selection().canCopy()) {
+            return;
+        }
+        ChordRest* firstCR = selection().firstChordRest();
+        if (firstCR->isGrace()) {
+            firstCR = toChordRest(firstCR->parent());
+        }
+        TDuration initialDuration = firstCR->ticks();
+        TDuration d = initialDuration.shiftRetainDots(nSteps, stepDotted);
+        if (!d.isValid()) {
+            return;
+        }
+        Fraction scale = d.ticks() / initialDuration.ticks();
+        for (ChordRest* cr : getSelectedChordRests()) {
+            Fraction newTicks = cr->ticks() * scale;
+            if (newTicks < Fraction(1, 1024)
+                || (stepDotted && cr->durationType().dots() != firstCR->durationType().dots()
+                    && !cr->isGrace())) {
+                return;
+            }
+        }
+        const muse::ByteArray mimeData(selection().mimeData());
+        XmlReader e(mimeData);
+        deleteRange(selection().startSegment(), selection().endSegment(), staff2track(selection().staffStart()),
+                    staff2track(selection().staffEnd()), selectionFilter());
+        pasteStaff(e, selection().startSegment(), selection().staffStart(), scale);
```

Comment:
I'm afraid the combination of `deleteRange` and `pasteStaff` may result in the same kind of trouble as we had in https://github.com/musescore/MuseScore/pull/27411. See the issue linked in that PR for what kind of trouble exactly, and see the second commit of that PR of how to resolve it.